spec/support/shared/unit/double_world.rb

```diff
@@ -0,0 +1,79 @@
+#
+# Author:: John Keiser <jkeiser@chef.io>
+# Copyright:: Copyright (c) 2015 John Keiser.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Mocks File and ShellOut.
+#
+# world.files["/etc/rc.d/init.d"] = "blah" means that /etc/rc.d/init.d is a file with contents "blah."
+# world.commands["shellme"] = "result" means that when shell_out! calls "shellme", it will return "result" in stdout.
+# world.commands["shellme"] = 10 means that when shell_out! calls "shellme", it will return exit code 10
+# world.commands["shellme"] = { code: 10, stdout: "result", stderr: "result2"}
+#
+shared_context "double world" do
+  let(:world) { DoubleWorld.new }
+  before do
+    allow(::File).to receive(:exist?)      { |f| world.files.has_key?(f) }
```

Comment:
I'd prefer a stricter approach where you have to say which files you expect the code to try to access that _don't_ exist. Maybe you would do `world.files["/etc/rc.d/init.d"] = false` or something like that. That way you'd have to be explicit about what the state of the filesystem is and you're less likely to accidentally rely on the implicit file-doesn't-exist assumption built in to this.